src/verify-event.ts

```diff
@@ -0,0 +1,95 @@
+import * as log from './log';
+import WeakMap from './weakmap';
+
+/**
+ * Gets the event that is being currently handled.
+ * @suppress {es5Strict}
+ */
+export function retrieveEvent():Event {
+    log.call('Retrieving event');
+    let currentEvent = window.top.event; // Attaching `window` to get around a bug of AG wrapper
+    if (!currentEvent) {
+        log.print('window.event does not exist, trying to get event from Function.caller');
+        try {
+            let caller = arguments.callee;
+            let touched = new WeakMap();
+            while (caller.caller) {
+                caller = caller.caller;
+                if (touched.has(caller)) {
+                    throw /* @ifdef DEBUG */ "Recursion in the call stack"; /* @endif */ /* @ifndef DEBUG */ null; /* @endif */
```

Comment:
1. Do not abuse the pre-compiler directives
2. Use debug for extended logging, but never change the workflow